### PR TITLE
Add Configuration for Kotlin's All-Open Plugin for JPA Entities

### PR DESCRIPTION
**Pull request by YangSiJun528**
~~Currently, the `KotlinJpaGradleBuildCustomizer` class is responsible for configuring either `javax` or `jakarta` dependencies. I believe a cleaner approach would be to split this functionality into two separate classes:~~
- ~~`KotlinJavaxJpaGradleBuildCustomizer`~~
- ~~`KotlinJakartaJpaGradleBuildCustomizer`~~

~~I'm unsure whether these new classes should inherit from `KotlinJpaGradleBuildCustomizer` or be designed as independent classes that implement `BuildCustomizer`. (similar to `KotlinJpaMavenBuildCustomizer`)~~

~~I would appreciate feedback on this approach.~~

The strikethrough was used because JavaEE support is no longer needed. This makes the previous concerns irrelevant.

Fixes gh-1572

### CODE REVIEW

**Comment by YangSiJun528** [comment]
I'm currently working on fixing several bugs in this code, so no need to review it right now. 

I'll tag you once everything is resolved. 
Feel free to close the PR, and I'll open a new one after the fixes.

File: initializr-generator-spring/src/main/java/io/spring/initializr/generator/spring/code/kotlin/KotlinJpaGradleBuildCustomizer.java [hunk 1]
```diff
@@ -46,7 +47,29 @@ public void customize(GradleBuild build) {
 		if (this.buildMetadataResolver.hasFacet(build, "jpa")) {
 			build.plugins()
 				.add("org.jetbrains.kotlin.plugin.jpa", (plugin) -> plugin.setVersion(this.settings.getVersion()));
+			if (this.buildMetadataResolver.hasGroupId(build, "jakarta.persistence")) {
+				customizeAllOpenWithJakarta(build);
+			}
+			else if (this.buildMetadataResolver.hasGroupId(build, "javax.persistence")) {
```
Comment (mhalbritter) [inline initializr-generator-spring/src/main/java/io/spring/initializr/generator/spring/code/kotlin/KotlinJpaGradleBuildCustomizer.java:53]: We don't need to support JavaEE anymore. Our baseline is Spring Boot 3.2, which already uses JakartaEE.

File: initializr-generator-spring/src/main/java/io/spring/initializr/generator/spring/code/kotlin/KotlinJpaMavenBuildCustomizer.java [hunk 2]
```diff
@@ -41,11 +43,31 @@ public KotlinJpaMavenBuildCustomizer(InitializrMetadata metadata, ProjectDescrip
...
 			build.plugins().add("org.jetbrains.kotlin", "kotlin-maven-plugin", (kotlinPlugin) -> {
-				kotlinPlugin.configuration((configuration) -> configuration.configure("compilerPlugins",
-						(compilerPlugins) -> compilerPlugins.add("plugin", "jpa")));
+				kotlinPlugin.configuration((configuration) -> {
+					configuration.configure("compilerPlugins",
+							(compilerPlugins) -> compilerPlugins.add("plugin", "jpa"));
+					if (this.buildMetadataResolver.hasGroupId(build, "jakarta.persistence")) {
+						customizeAllOpenWithJakarta(configuration);
+					}
+					else if (this.buildMetadataResolver.hasGroupId(build, "javax.persistence")) {
```
Comment (mhalbritter) [inline initializr-generator-spring/src/main/java/io/spring/initializr/generator/spring/code/kotlin/KotlinJpaMavenBuildCustomizer.java:52]: We don't need to support JavaEE anymore.

File: initializr-generator-spring/src/test/java/io/spring/initializr/generator/spring/code/kotlin/KotlinJpaGradleBuildCustomizerTests.java [hunk 3]
```diff
@@ -58,6 +61,41 @@ void customizeWhenJpaFacetAbsentShouldNotAddKotlinJpaPlugin() {
...
+			assertThat(extension.getInvocations())
+				.filteredOn((invocation) -> Objects.equals(invocation.getTarget(), "annotation"))
+				.extracting("arguments")
+				.containsExactlyInAnyOrder(List.of("jakarta.persistence.Entity"),
+						List.of("jakarta.persistence.MappedSuperclass"), List.of("jakarta.persistence.Embeddable"));
+		});
+	}
+
+	@Test
+	void customizeWhenJavaxPersistencePresentShouldCustomizeAllOpenWithJavax() {
```
Comment (mhalbritter) [inline initializr-generator-spring/src/test/java/io/spring/initializr/generator/spring/code/kotlin/KotlinJpaGradleBuildCustomizerTests.java:82]: We don't need to support JavaEE anymore.

File: initializr-generator-spring/src/test/java/io/spring/initializr/generator/spring/code/kotlin/KotlinJpaMavenBuildCustomizerTests.java [hunk 4]
```diff
@@ -31,12 +31,6 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-/**
```
Comment (mhalbritter) [inline initializr-generator-spring/src/test/java/io/spring/initializr/generator/spring/code/kotlin/KotlinJpaMavenBuildCustomizerTests.java:34]: Why did you remove this?

File: initializr-generator-spring/src/test/java/io/spring/initializr/generator/spring/code/kotlin/KotlinJpaMavenBuildCustomizerTests.java [hunk 5]
```diff
@@ -67,6 +61,48 @@ void customizeWhenJpaFacetAbsentShouldNotAddKotlinJpaPlugin() {
...
+				.filter((setting) -> "option".equals(setting.getName()))
+				.map(MavenPlugin.Setting::getValue))
+				.containsExactlyInAnyOrder("all-open:annotation=jakarta.persistence.Entity",
+						"all-open:annotation=jakarta.persistence.MappedSuperclass",
+						"all-open:annotation=jakarta.persistence.Embeddable");
+		});
+	}
+
+	@Test
+	void customizeWhenJavaxPersistencePresentShouldCustomizeAllOpenWithJavax() {
```
Comment (mhalbritter) [inline initializr-generator-spring/src/test/java/io/spring/initializr/generator/spring/code/kotlin/KotlinJpaMavenBuildCustomizerTests.java:86]: No need to support JavaEE anymore.

**Review by mhalbritter — COMMENTED** [review]
Hey @YangSiJun528,

thanks for the PR. I've left some comments.

**Comment by YangSiJun528** [comment]
I have fixed the bug and incorporated your feedback, @mhalbritter :
- Restored the accidentally deleted comments.
- Removed the support for JavaEE.

**Comment by mhalbritter** [comment]
Thanks @YangSiJun528 !
